src/voting/rankedChoice.ts

```diff
@@ -148,16 +187,57 @@ export default class RankedChoiceVoting {
     );
   }
 
+  /**
+   * Returns the total voting power from all submitted votes, including invalid ones.
+   *
+   * This method sums the balance (voting power) from ALL votes submitted to the proposal,
+   * regardless of whether they have valid choice arrays. This is useful for calculating
+   * total participation, quorum requirements, and percentage of total voting power.
+   *
+   * Note: This differs from IRV final results which only include valid votes. Invalid votes
+   * are excluded from IRV calculations but their voting power is still counted here for
+   * participation metrics.
+   *
+   * @returns Total voting power from all votes (valid + invalid)
+   *
+   * @example
+   * // votes = [
+   * //   { choice: [1,2,3,4], balance: 1000 }, // Valid
+   * //   { choice: [1,5,2], balance: 500 },    // Invalid (index 5)
+   * //   { choice: [2,1,4,3], balance: 750 }   // Valid
+   * // ]
+   * // Returns: 2250 (includes invalid vote's 500 balance)
+   */
   getScoresTotal(): number {
     return this.votes.reduce((a, b: any) => a + b.balance, 0);
   }
 
+  /**
+   * Converts the selected choice indices into a human-readable string representation.
+   *
+   * Note: This method supports partial ranking where not all available choices
+   * need to be selected. The ordinal positions (1st, 2nd, etc.) reflect the
+   * order of valid selections only. Invalid choice indices are filtered out.
+   *
+   * @returns A formatted string showing the ranked choices with ordinal positions.
+   * Only valid choices are included, invalid indices are silently ignored.
+   *
+   * @example
+   * // With choices ['Alice', 'Bob', 'Carol', 'David'] and selected [1, 3, 2]
+   * // Returns: "(1st) Alice, (2nd) Carol, (3rd) Bob"
+   *
+   * @example
+   * // Partial ranking with choices ['Alice', 'Bob', 'Carol', 'David'] and selected [4, 1]
+   * // Returns: "(1st) David, (2nd) Alice"
+   *
+   * @example
+   * // With invalid choice index 5 in selected [1, 5]
+   * // Returns: "(1st) Alice" - invalid choice 5 is filtered out
+   */
   getChoiceString(): string {
     return this.selected
-      .map((choice) => {
-        if (this.proposal.choices[choice - 1])
-          return this.proposal.choices[choice - 1];
-      })
+      .map((choice) => this.proposal.choices[choice - 1])
+      .filter(Boolean)
```

Comment:
Should be fine with reduced array length?

Reply:
We are filtering some choices here (before we are not filtering) so there will be difference in array length, i think it's fine but want to double check because PR is just refactor 😅

Reply:
This is a little inconsistency I found while writing tests.

We are filtering out invalid choices everywhere, but single choice.

In which case this can break ? I don't think we check for arrays length anywhere before calling this function

Reply:
Okies